db/schema.rb

```diff
@@ -1449,4 +1449,91 @@
+              count(DISTINCT reporting_patient_states.patient_id) FILTER (WHERE ((reporting_patient_states.htn_treatment_outcome_in_last_3_months = 'visited_no_bp'::text) AND (reporting_patient_states.htn_care_state = 'under_care'::text))) AS visited_no_bp_under_care,
+              count(DISTINCT reporting_patient_states.patient_id) FILTER (WHERE ((reporting_patient_states.htn_treatment_outcome_in_last_3_months = 'visited_no_bp'::text) AND (reporting_patient_states.htn_care_state = 'lost_to_follow_up'::text))) AS visited_no_bp_lost_to_follow_up,
+              count(DISTINCT reporting_patient_states.patient_id) FILTER (WHERE (reporting_patient_states.htn_care_state = 'under_care'::text)) AS patients_under_care,
+              count(DISTINCT reporting_patient_states.patient_id) FILTER (WHERE (reporting_patient_states.htn_care_state = 'lost_to_follow_up'::text)) AS patients_lost_to_follow_up
+             FROM reporting_patient_states
+            WHERE ((reporting_patient_states.hypertension = 'yes'::text) AND (reporting_patient_states.months_since_registration >= (3)::double precision))
+            GROUP BY reporting_patient_states.assigned_facility_region_id, reporting_patient_states.month_date
+          )
+   SELECT cal.month_date,
+      cal.month,
+      cal.quarter,
+      cal.year,
+      cal.month_string,
+      cal.quarter_string,
+      rf.facility_id,
+      rf.facility_name,
+      rf.facility_type,
+      rf.facility_size,
+      rf.facility_region_id,
+      rf.facility_region_name,
+      rf.facility_region_slug,
+      rf.block_region_id,
+      rf.block_name,
+      rf.block_slug,
+      rf.district_id,
+      rf.district_region_id,
+      rf.district_name,
+      rf.district_slug,
+      rf.state_region_id,
+      rf.state_name,
+      rf.state_slug,
+      rf.organization_id,
+      rf.organization_region_id,
+      rf.organization_name,
+      rf.organization_slug,
+      rp.cumulative_registrations,
+      rp.monthly_registrations,
+      ap.under_care,
+      ap.lost_to_follow_up,
+      ap.dead,
+      ap.assigned_patients,
+      tout.controlled_under_care,
+      tout.controlled_lost_to_follow_up,
+      tout.uncontrolled_under_care,
+      tout.uncontrolled_lost_to_follow_up,
+      tout.missed_visit_under_care,
+      tout.missed_visit_lost_to_follow_up,
+      tout.visited_no_bp_under_care,
+      tout.visited_no_bp_lost_to_follow_up,
+      tout.patients_under_care,
+      tout.patients_lost_to_follow_up
+     FROM ((((registered_patients rp
+       LEFT JOIN assigned_patients ap ON (((rp.region_id = ap.region_id) AND (rp.month_date = ap.month_date))))
+       LEFT JOIN treatment_outcomes_in_last_3_months tout ON (((rp.region_id = tout.region_id) AND (rp.month_date = tout.month_date))))
+       JOIN reporting_facilities rf ON ((rp.region_id = rf.facility_region_id)))
+       JOIN reporting_months cal ON ((rp.month_date = cal.month_date)));
```

Comment:
These were the only changes here -- we have to use `LEFT JOIN` on assigned and treatment outcomes, otherwise we would lose records for brand new facilities where they _only_ have patients registered but no assigned patients. This happened in a bunch of tests, which admittedly may not represent how prod data looks...but we've seen enough strange edge cases happen when new facilities start that I wouldn't be surprised if this could be a real thing.

Reply:
Makes sense, @rsanheim. I've made a change with fe2501e6968a8b23864e88fedb3cf2d379bc17bb that ensures that we will have a row for every facility and month combination. Even if a bit redundant, I feel this is simple and creates a certainty on the rows present in the table.